### PR TITLE
Add copy release file download link button

### LINKED ISSUE #1275: Add ability to copy release file download link
I want to use a different downloader, so I'd like to be able to copy the download URL for a file. This could be implemented by holding down on the file, or adding a copy link button.

### PR DESCRIPTION
**Pull request by Vethya**
Resolves #1275

### CODE REVIEW

File: app/src/main/java/com/gh4a/adapter/ReleaseAssetAdapter.java [hunk 1]
```diff
@@ -33,12 +37,25 @@ public void onBindViewHolder(ViewHolder holder, ReleaseAsset asset) {
...
                 StringUtils.formatRelativeTime(mContext, asset.createdAt(), true)));
         holder.tvSize.setText(Formatter.formatFileSize(mContext, asset.size()));
         holder.tvDownloads.setText(String.valueOf(asset.downloadCount()));
     }
 
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.btn_copy_link) {
+            ReleaseAsset asset = (ReleaseAsset) v.getTag();
+            String label = "Release asset " + asset.name();
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/adapter/ReleaseAssetAdapter.java:52]: This needs to be a string resource.

File: app/src/main/res/layout/row_download.xml [hunk 2]
```diff
@@ -41,6 +41,20 @@
...
     </LinearLayout>
 
+    <ImageButton
+        android:id="@+id/btn_copy_link"
+        android:layout_width="50dp"
+        android:layout_height="50dp"
+        style="@style/SelectableBorderlessItem"
+        android:layout_gravity="center_vertical"
+        android:contentDescription="@string/copy_download_link"
+        android:src="@drawable/icon_copy_link" />
```
Comment (maniac103) [inline app/src/main/res/layout/row_download.xml:51]: I think I'd rather add this as long.-press action. Long press for copying links is a convention I've often seen in other apps, and doing so makes the asset item less visually crowded.

**Review by the-blank-x — COMMENTED** [review]
Perhaps a toast can be shown [on Android versions under 13](https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications) to let the user know that the asset link is copied?

File: app/src/main/java/com/gh4a/activities/ReleaseInfoActivity.java [hunk 3]
```diff
@@ -234,6 +238,16 @@ public void onItemClick(ReleaseAsset item) {
         DownloadUtils.enqueueDownloadWithPermissionCheck(this, item);
     }
 
+    @Override
+    public boolean onItemLongClick(ReleaseAsset item) {
+        String label = "Release asset " + item.name();
+        IntentUtils.copyToClipboard(this, label, item.browserDownloadUrl());
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2)
+            Toast.makeText(this, R.string.link_copied, Toast.LENGTH_SHORT).show();
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/activities/ReleaseInfoActivity.java:246]: I'd rather do this:

```
Snackbar.make(getRootLayout(), R.string.link_copied, Snackbar.LENGTH_SHORT).show();
```

**Pull request merged**